tests/robot-tests/tests/libs/slack.py

```diff
@@ -0,0 +1,86 @@
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+import shutil
+import json
+import requests
+from bs4 import BeautifulSoup
+
+PATH = f'{os.getcwd()}{os.sep}test-results'
+
+
+def _generate_slack_attachments():
+    failed_tests = dict()
+    passed_tests = dict()
+
+    report = open(f'{PATH}{os.sep}output.xml', 'rb')
+    contents = report.read()
+
+    soup = BeautifulSoup(contents, 'xml')
+    test = soup.find('total').find('stat')
+
+    failed_tests = test['fail']
+    passed_tests = test['pass']
+
+    return [
+        {
+            "pretext": "All results",
+            "color": "danger" if failed_tests else "good",
+            "mrkdwn_in": [
+                "pretext"
+            ],
+            "fields": [
+                {
+                    "title": "Total test cases",
+                    "value": int(passed_tests) + int(failed_tests)
+                },
+                {
+                    "title": "Passed tests",
+                    "value": int(passed_tests)
+                },
+                {
+                    "title": "Failed tests",
+                    "value": int(failed_tests) if failed_tests else "0"
+                },
+                {
+                    "title": "Results",
+                    "value": "Failed" if failed_tests else "Passed"
+                },
+
+            ]
+        }
+    ]
+
+
+def send_slack_report():
+    attachments = _generate_slack_attachments()
+    data = {"attachments": attachments}
+
+    webhook_url = os.getenv('SLACK_TEST_REPORT_WEBHOOK_URL')
+    slack_bot_token = os.getenv('SLACK_BOT_TOKEN')
+
+    assert webhook_url, print("SLACK_TEST_REPORT_WEBHOOK_URL env variable needs to bet set")
+    assert slack_bot_token, print("SLACK_BOT_TOKEN env variable needs to bet set")
+
+    response = requests.post(
+        url=webhook_url,
+        data=json.dumps(data), headers={'Content-Type': 'application/json'}
+    )
+    assert response.status_code == 200, print(
+        "Response wasn't 200, it was: {}".format({response}))
```

Comment:
It's not a big deal, but I'd switch using an f-string here to be consistent with how we do things elsewhere:
`f"Response wasn't 200, it was {response}"`